spec/support/request.rb

```diff
@@ -6,7 +6,9 @@ class SessionHelperController < ApplicationController
       respond_to :json
 
       def update
-        session[params[:key]] = params[:value]
+        value = params.require(:value)
+        value = value.to_unsafe_hash if value.respond_to?(:to_unsafe_hash)
+        session[params[:key]] = value
```

Comment:
These changes are required due to the serialization capabilities. `params[:value]` was returning an instance of ActionController::Parameters and it was stored and returned from the session.